runtime/interpreter/errors.go

```diff
@@ -239,6 +239,9 @@ func (e RedeclarationError) Error() string {
 // DereferenceError
 
 type DereferenceError struct {
+	Cause        string
+	ExpectedType *sema.Type
+	ActualType   *sema.Type
```

Comment:
These fields can be non-pointers (`sema.Type` is an interface)

Reply:
I made these pointer types because these fields are not always included in the error struct; i.e. they can be `nil`. If these aren't pointer types won't we run afoul of the dynamic-`nil` type behavior that has caused issues in the past?

Reply:
No, not in this case. We do not have typed-nil values for `sema.Type`, and also use `Cause` to decide if `ExpectedType` and `ActualType` should be used / are set, or if not; instead of checking if those fields are  `nil`